src/libPMacc/include/cuSTL/container/allocator/compile-time/SharedMemAllocator.hpp

```diff
@@ -73,11 +97,23 @@ struct SharedMemAllocator<Type, Size, 3, uid>
     BOOST_STATIC_CONSTEXPR int dim = 3;
     typedef cursor::CT::BufferCursor<type, Pitch> Cursor;
 
-    __device__ static Cursor allocate()
+protected:
+
+    Cursor cursor;
+
+    HDINLINE SharedMemAllocator() : cursor(NULL) {}
+
+    /* has to be static, unless a crash is acceptable */
+    DINLINE static Cursor _allocate()
```

Comment:
Please [avoid prefix `_` ](http://stackoverflow.com/questions/228783/what-are-the-rules-about-using-an-underscore-in-a-c-identifier)